### PR TITLE
Write to transport \r\n before closing after keepalive timeout #1883

### PR DESCRIPTION
**Pull request by fafhrd91**

### CODE REVIEW

**Comment by fafhrd91** [comment]
@kxepal @asvetlov comments?

File: aiohttp/web_protocol.py [hunk 1]
```diff
@@ -268,6 +268,7 @@ def eof_received(self):
         pass
 
     def data_received(self, data):
+        print("data:", data)
```
Comment (kxepal) [inline aiohttp/web_protocol.py]: print

File: aiohttp/web_protocol.py [hunk 2]
```diff
@@ -336,13 +337,15 @@ def close(self):
             if not waiter.done():
                 waiter.cancel()
 
-    def force_close(self):
+    def force_close(self, keepalive=False):
```
Comment (kxepal) [inline aiohttp/web_protocol.py]: I think the argment name a bit misleading. `send_last_heartbeat` - might be more precisely about what will happen. The `keepalive` fits more to class property which can be enabled or disabled.

Comment (fafhrd91) [reply aiohttp/web_protocol.py]: good catch, thanks

**Pull request closed without merging**